File: programmers/biggestnum.cpp
#include <vector>
#include <string>
#include <algorithm>
using namespace std;
 
bool comp(string a, string b) {
    return a + b > b + a;
}
 
string solution(vector<int> numbers) {
    string answer = "";
    vector<string> numstr;

    for (auto i : numbers) numstr.push_back(to_string(i));

    sort(numstr.begin(), numstr.end(), comp);

    if (numstr.at(0) == "0") return "0";
    for (auto i : numstr)
        answer += i;
    return answer;
}

File: programmers/biggestnum.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        
        String [] str = new String[numbers.length];
        
        for(int i = 0; i < numbers.length; i++){
            str[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(str, new Comparator<String>(){
            @Override
            public int compare(String a, String b){
                return (b+a).compareTo(a+b);
            }
        });
        
        if (str[0].equals("0")) return "0";
        
        for(String s: str) answer += s;        
        
        return answer;
    }
}
